activerecord/test/cases/adapters/postgresql/enum_test.rb

```diff
@@ -139,4 +139,21 @@ def test_works_with_activerecord_enum
     model = PostgresqlEnum.find(model.id)
     assert model.current_mood_happy?
   end
+
+  def test_enum_type_scoped_to_schemas
+    @connection.create_schema("test_schema")
+    @connection.schema_search_path = "test_schema"
+    @connection.schema_cache.clear!
+    @connection.create_enum("mood", ["sad", "ok", "happy"])
+
+    assert_nothing_raised do
+      @connection.create_table("postgresql_enums") do |t|
+        t.column :current_mood, :mood, default: "happy", null: false
+      end
+    end
+  ensure
+    @connection.drop_schema "test_schema"
+    @connection.schema_search_path = "'$user', public"
```

Comment:
Rather than hard coding the value here I think you should store the current value at the start of the test and reset it here.

Reply:
Done. Thanks.